assimilation_code/modules/assimilation/filter_mod.f90

```diff
@@ -2719,6 +2730,69 @@ subroutine set_copies(cnum, STAGE_COPIES)
 
 end subroutine set_copies
 
+
+!-------------------------------------------------------------------------------
+!> The infl_flavor namelist is a string, and specifies the inflation algorithm.
+!> The character string can either be the name associated with the type of inflation
+!> or the integer associated with the type of inflation. The string names of the
+!> inflation algorithms is based on what is declared in the adaptive_inflate_mod.f90
+!> which is repeated here for reference. 
+!>
+!> NO_INFLATION               = 0
+!> OBS_INFLATION              = 1    observation-space inflation (deprecated)
+!> VARYING_SS_INFLATION       = 2    spatially-varying state-space inflation
+!> SINGLE_SS_INFLATION        = 3    spatially-constant state-space inflation
+!> RELAXATION_TO_PRIOR_SPREAD = 4    (available only with posterior inflation)
+!> ENHANCED_SS_INFLATION      = 5    Inverse Gamma version of VARYING_SS_INFLATION
```

Comment:
tim, can we talk about whether it's feasible to move some of this routine into the utilities module so we don't end up replicating this code (upper case, select, etc) in every module we want to make this change?  maybe it's impractical but it's worth a conversation, i think.

Reply:
I ran some tests as to what is valid input for a variable defined to be a character(len=?) ... which surprised me a little.  I always knew that if you wanted to include a slash '/' in a character variable, you had to protect the entire string with quotes to prevent the namelist parser from thinking the slash was the end of the namelist. Seems gfortran requires alphabetic characters to be quoted, while SINGLE numeric characters do not need to be quoted.  Examples of syntaxes that work:

```
   inf_flavor                  = 0,          'RTPS'
   inf_flavor                  = '0',          'RTPS'
   inf_flavor                  = 'NO_INFLATION',          'RTPS'
   inf_flavor                  = 'NO_INFLATION',          0
```
and some syntaxes that fail:
```
   inf_flavor                  = 0,          RTPS,
   inf_flavor                  = 00,          0,
   inf_flavor                  = 'NO_INFLATION',          'RTPS'
```
Not sure if that is definitive enough to answer the question about whether the comma is a sufficient delimeter ...

Reply:
I created   utilities_mod:flex_parser() that does what I understood from my talk with Nancy. Hopefully, this is along the lines of what she meant!